arkane/explorer.py

```diff
@@ -143,10 +143,19 @@ def execute(self, outputFile, plot, format='pdf', print_summary=True, speciesLis
                     self.source[i] = spc
         
         # react initial species
-        flags = np.array([s.molecule[0].getFormula()==form for s in reaction_model.core.species])
+        if len(self.source) == 1:
+            flags = np.array([s.molecule[0].getFormula()==form for s in reaction_model.core.species])
+            biflags = np.zeros((len(reaction_model.core.species),len(reaction_model.core.species)))
+        elif len(self.source) == 2:
+            flags = np.array([s.molecule[0].getFormula()==form for s in reaction_model.core.species])
```

Comment:
If `flags` is being set the same regardless of the condition, perhaps set it before the if statement? Or should it be set to zeroes if `len(self.source)==2`?

Reply:
Fixed